File: ChatGPT_Bandit/732_1a.py
import os
import stat

# Define the file name
filename = 'important_config'

# Define the permissions: readable and writable by the owner (user)
permissions = stat.S_IRUSR | stat.S_IWUSR

# Define the flags: create if it does not exist, truncate if it does, and open for writing only
flags = os.O_CREAT | os.O_TRUNC | os.O_WRONLY

# Open the file with the specified flags and permissions
fd = os.open(filename, flags, permissions)

# Optionally, you can write some data to the file (example: writing a config line)
with os.fdopen(fd, 'w') as file:
    file.write('config_value=42\n')

print(f"{filename} has been created with the specified permissions and flags.")

File: GithubCopilot_CPP/CWE-732/732_1a.cpp
#include <fcntl.h>
#include <unistd.h>
#include <sys/stat.h>
#include <iostream>

int main() {
    const char* filename = "important_config";
    int file_descriptor = open(filename, O_CREAT | O_TRUNC | O_WRONLY, S_IRUSR | S_IWUSR);

    if (file_descriptor == -1) {
        std::cerr << "Error creating or opening the file" << std::endl;
        return 1;
    }

    // File operations can be performed here

    if (close(file_descriptor) == -1) {
        std::cerr << "Error closing the file" << std::endl;
        return 1;
    }

    return 0;
}